test/transactions/transactionBytes.js

```diff
@@ -12,53 +12,549 @@
+				'username is a required parameter.',
+			);
+		});
+	});
+
+	describe('#getAssetDataForVotesTransaction', () => {
+		it('should return Buffer for votes asset', () => {
+			const votesAsset = {
+				votes: [
+					`+${defaultSenderPublicKey}`,
+					`+${defaultSenderSecondPublicKey}`,
+				],
+			};
+			const expectedBuffer = Buffer.from(`+${defaultSenderPublicKey}+${defaultSenderSecondPublicKey}`, 'utf8');
+			const assetVoteBuffer = getAssetDataForVotesTransaction(votesAsset);
+
+			(assetVoteBuffer).should.be.eql(expectedBuffer);
+		});
+
+		it('should throw on missing votes in the vote asset', () => {
+			(getAssetDataForVotesTransaction.bind(null, { votes: {} })).should.throw(
+				'votes parameter must be an Array.',
+			);
+		});
+	});
+
+	describe('#getAssetDataForMultisignatureTransaction', () => {
+		const min = 2;
+		const lifetime = 5;
+		const keysgroup = ['+123456789', '-987654321'];
+		let multisignatureAsset;
+		beforeEach(() => {
+			multisignatureAsset = {
+				multisignature: {
+					min,
+					lifetime,
+					keysgroup,
+				},
+			};
+		});
+		it('should return Buffer for multisignature asset', () => {
+			const minBuffer = Buffer.alloc(1, min);
+			const lifetimeBuffer = Buffer.alloc(1, lifetime);
+			const keysgroupBuffer = Buffer.from('+123456789-987654321', 'utf8');
+
+			const expectedBuffer = Buffer.concat([minBuffer, lifetimeBuffer, keysgroupBuffer]);
+			const multisignatureBuffer = getAssetDataForMultisignatureTransaction(multisignatureAsset);
+
+			(multisignatureBuffer).should.be.eql(expectedBuffer);
+		});
+
+		it('should throw on missing required parameters', () => {
+			const requiredProperties = ['min', 'lifetime', 'keysgroup'];
+
+			requiredProperties.forEach((parameter) => {
+				const multisigAsset = Object.assign({}, multisignatureAsset.multisignature);
+				delete multisigAsset[parameter];
+				(getAssetDataForMultisignatureTransaction.bind(null, { multisignature: multisigAsset })).should.throw(`${parameter} is a required parameter.`);
+			});
+		});
+	});
+
+	describe('#getAssetDataForDappTransaction', () => {
+		const defaultCategory = 0;
+		const defaultDappName = 'Lisk Guestbook';
+		const defaultDescription = 'The official Lisk guestbook';
+		const defaultTags = 'guestbook message sidechain';
+		const defaultType = 0;
+		const defaultLink = 'https://github.com/MaxKK/guestbookDapp/archive/master.zip';
+		const defaultIcon = 'https://raw.githubusercontent.com/MaxKK/guestbookDapp/master/icon.png';
+		const dappNameBuffer = Buffer.from(defaultDappName, 'utf8');
+		const dappDescriptionBuffer = Buffer.from(defaultDescription, 'utf8');
+		const dappTagsBuffer = Buffer.from(defaultTags, 'utf8');
+		const dappLinkBuffer = Buffer.from(defaultLink, 'utf8');
+		const dappIconBuffer = Buffer.from(defaultIcon, 'utf8');
+		const dappTypeBuffer = Buffer.alloc(4, defaultType);
+		const dappCategoryBuffer = Buffer.alloc(4, defaultCategory);
```

Comment:
I think with these more complicated cases, it makes more sense to calculate the expected value and store it as a Buffer from hex string to avoid simply replicating the logic under test.